### PR TITLE
Support podOverrides

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

For https://github.com/stackabletech/issues/issues/346

Try with
```
---
apiVersion: trino.stackable.tech/v1alpha1
kind: TrinoCluster
metadata:
  name: simple-trino
spec:
  image:
    productVersion: "414"
    stackableVersion: "0.0.0-dev"
  clusterConfig:
    catalogLabelSelector: {}
  coordinators:
    podOverrides: # on role level
      spec:
        tolerations:
          - key: "key1"
            operator: "Equal"
            value: "value1"
            effect: "NoSchedule"
    roleGroups:
      default:
        podOverrides: # on rolegroup level, takes precedence over the ones specified at role level
          metadata:
            labels:
              my-custom-label: super-important-label
        replicas: 1
  workers:
    roleGroups:
      default:
        replicas: 1
```

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -3,5 +3,6 @@ members = [
     "rust/crd", "rust/operator-binary"
 ]
 
-#[patch."https://github.com/stackabletech/operator-rs.git"]
-#stackable-operator = { git = "https://github.com/stackabletech//operator-rs.git", branch = "main" }
+[patch."https://github.com/stackabletech/operator-rs.git"]
+stackable-operator = { git = "https://github.com/stackabletech//operator-rs.git", branch = "feat/pod-overrides" }
```
Comment (nightkr) [inline Cargo.toml:7]: Reminder to revert this once updated

**Review by nightkr — APPROVED** [review]
Haven't tested but LGTM in theory

Remember the changelog!

**Comment by sbernauer** [comment]
@nightkr ready to be merged, would be great if you could have another look

File: rust/operator-binary/src/controller.rs [hunk 2]
```diff
@@ -925,6 +926,12 @@ fn build_rolegroup_statefulset(
             "config",
             "log",
             merged_config.logging.containers.get(&Container::Vector),
+            ResourceRequirementsBuilder::new()
```
Comment (nightkr) [inline rust/operator-binary/src/controller.rs:929]: I'd say this deserves a changelog callout as well

**Review by nightkr — APPROVED** [review]
Looks good in theory, haven't tested it myself

File: rust/operator-binary/src/controller.rs (hunk 2 above)
Comment (sbernauer) [reply rust/operator-binary/src/controller.rs:929]: Agreed! Let's way till we decided if we are going to merge podOverrides or the whole resource stuff first

Comment (nightkr) [reply rust/operator-binary/src/controller.rs:929]: Is it viable to split this out into a separate PR?

Comment (sbernauer) [reply rust/operator-binary/src/controller.rs:929]: Some context: This is a breaking change in operator-rs (the resources are now required), which we decided to release together with the resource PR. So I *need* to explicitly specify the resources.
However, after thinking about this, I would simply re-use the default we had previously in operator-rs, to *not* change the behavior in any way. So this PR can be completely independent of the resources PR that will follow.

WDYT?

Comment (sbernauer) [reply rust/operator-binary/src/controller.rs:929]: Nevermind, we never had resources set for the vector container, so we need to decide on some value now

Comment (sbernauer) [reply rust/operator-binary/src/controller.rs:931]: Updated the resources to the value we agreed this morning

Comment (nightkr) [reply rust/operator-binary/src/controller.rs:931]: Who's "we" in this context? Especially the inline comment is pretty useless and could do with a reference back to that discussion.

Comment (sbernauer) [reply rust/operator-binary/src/controller.rs:931]: Maxi, Razvan, Sasch, Jim, Malte and me (I might missed someone) in the meeting this morning at 09. Anyway, I know this was going to happen when we stuff everything in a single operator-rs release :upside_down_face: 
Let's wait till the resource PRs have been merged before continuing with podOverrides

**Comment by sbernauer** [comment]
Superseded by https://github.com/stackabletech/trino-operator/pull/440

**Pull request closed without merging**